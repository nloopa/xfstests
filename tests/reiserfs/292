#! /bin/bash
# FS QA Test No. 292
#
# Ensure mkfs with stripe geometry goes into multidisk mode
# which results in more AGs
#
#-----------------------------------------------------------------------
# Copyright (c) 2012 Red Hat, Inc.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
    cd /
    rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

# real QA test starts here

# Modify as appropriate.
_supported_fs xfs
_supported_os IRIX Linux

fsfile=$TEST_DIR/fsfile.$seq

rm -f $fsfile
$XFS_IO_PROG -f -c "truncate 256g" $fsfile

echo "mkfs.xfs without geometry"
mkfs.xfs -f $fsfile 2>&1 | sed -e "s:$fsfile:FILENAME:g" | grep meta-data
echo "mkfs.xfs with cmdline geometry"
mkfs.xfs -f -d su=16k,sw=5 $fsfile 2>&1 | sed -e "s:$fsfile:FILENAME:g" | grep meta-data

# success, all done
status=0
exit
