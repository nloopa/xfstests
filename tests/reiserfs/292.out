QA output created by 292
mkfs.xfs without geometry
meta-data=FILENAME   isize=256    agcount=4, agsize=16777216 blks
mkfs.xfs with cmdline geometry
meta-data=FILENAME   isize=256    agcount=16, agsize=4194304 blks
